hail/src/main/scala/is/hail/asm4s/Code.scala

```diff
@@ -1239,6 +1239,12 @@ object FieldRef {
   }
 }
 
+object Value {
+  def fromLIR[T](v: lir.ValueX): Value[T] = new Value[T] {
```

Comment:
I think this isn't valid -- the node `v` will appear in the LIR tree by object identity each time you `get` it, which is a violation of our data model.

Reply:
(am guessing this is what's causing some of your test failures)

Reply:
Ah, yup. Thanks!